### PR TITLE
Bug 1584321 - merge static_clients into dev-config.yml automatically

### PR DESCRIPTION
**Pull request by djmitche**
Merging static_clients using Helm is not possible, but we can at least
verify that the dev-config.yml has the correct scopes..

Bugzilla Bug: [1584321](https://bugzilla.mozilla.org/show_bug.cgi?id=1584321)

### CODE REVIEW

**Comment by djmitche** [comment]
Oh hang on, I bet I can make this work..

**Comment by djmitche** [comment]
This should help with dev deployments.  @imbstack how does this work with cloudops' deployments?

File: infrastructure/tooling/src/dev/helm.js [hunk 1]
```diff
@@ -22,9 +29,26 @@ const actions = [
       if (!config.meta || !config.meta.deploymentPrefix) {
         throw new Error('Must have configured dev-config.yml to deploy.');
       }
+
+      if (config.auth && config.auth.static_clients) {
+        if (config.auth.static_clients.some(({scopes}) => Boolean(scopes))) {
+          throw new Error('auth.static_clients in `dev-config.yml` should not contain scopes');
```
Comment (imbstack) [inline infrastructure/tooling/src/dev/helm.js]: Thanks for this (also just a good opportunity to use the multiline comment)

File: infrastructure/tooling/src/dev/helm.js [hunk 2]
```diff
@@ -22,9 +29,26 @@ const actions = [
...
     },
   },
+  {
+    title: 'Get static clients',
+    requires: ['dev-config'],
+    provides: ['static-clients'],
+    run: async (requirements, utils) => {
+      const ymlFile = await readRepoYAML('infrastructure/tooling/static-clients.yml');
+      const config = requirements['dev-config'];
+      const staticClients = await jsone(ymlFile, {azureAccountId: config.azureAccountId || 'unknown'});
```
Comment (imbstack) [inline infrastructure/tooling/src/dev/helm.js]: Do we actually want to fallback to `unknown` here or just bail?

Comment (djmitche) [reply infrastructure/tooling/src/dev/helm.js]: I don't know if this gets run in a case where this value isn't yet determined.  If not, I can remove the fallback.

**Comment by djmitche** [comment]
> I think they'll still need to update manually.

We should probably either make copy-pasting the static clients a standard part of the deployment process for cloudops (regardless of changelog), or find a way to automate it.

**Comment by djmitche** [comment]
In favor of #1494..

**Pull request closed without merging**